openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -129,6 +130,8 @@ def get_object(self):
         overview.user_has_access = overview.can_load_courseware
         overview.user_has_staff_access = overview.is_staff
 
+        # TODO: TNL-7185 Legacy: Refactor to return the expiration date and format the message in the MFE
+        overview.course_expired_message = generate_course_expired_message(self.request.user, overview)
```

Comment:
In the case that `generated_course_expired_message` returns none, will `course_expired_message = serializers.CharField()` have any [problems with an empty value](https://www.django-rest-framework.org/api-guide/fields/#charfield)?

Reply:
> Signature: CharField(..., allow_blank=False, ...)

> allow_blank - If set to True then the empty string should be considered a valid value. If set to False then the empty string is considered invalid and will raise a validation error. Defaults to False.

Weird... Reading those docs, I'd expect it to break, but that doesn't seem to be the case in my testing...

I hard-coded that function to return various values:
- Implicit and explicit `None` return values (`return` vs `return None`) render as `null` in the serialized JSON.
- empty string renders as an empty string in the serialized JSON.

Curious, I checked the source (suspecting maybe that default changed in a recent-ish version), but it appears to have been using the same defualt value for 6 years now:
- https://github.com/encode/django-rest-framework/blame/master/rest_framework/fields.py#L805

Reply:
Per discussion with @abutterworth , since this works even when returning `None`, we're going to merge as-is.